### PR TITLE
Common: add management of plugin sources

### PR DESCRIPTION
**Pull request by nglgzz**
### Please check if the PR fulfills these requirements
- [x] The commit message follows our [guidelines](https://github.com/neo4j-devtools/relate/blob/master/CONTRIBUTING.md)
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

### What kind of change does this PR introduce?
Feature

### What is the current behavior?
No logic is available for DBMS plugins.

### What is the new behavior?
Added `plugins` entity to environments and implemented methods for adding/removing/listing/getting plugin sources. 

### Does this PR introduce a breaking change?
No.

### Other information:
I stubbed the methods for managing DBMS plugins and only implemented the methods for managing plugin sources, as for now I'm mostly interested in feedback about the naming and structure of these methods. This work will be continued in later PRs.

### CODE REVIEW

File: packages/common/src/entities/plugins/plugins.local.ts [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        const sources = responses.mapEach((res) => new PluginSourceModel(res));
+
+        await sources
+            .mapEach(async (source) => {
+                const sourcePath = path.join(
+                    this.environment.dirPaths.pluginSources,
+                    `${source.name}${JSON_FILE_EXTENSION}`,
+                );
+
+                await fse.writeJSON(sourcePath, source);
```
Comment (huboneo) [inline packages/common/src/entities/plugins/plugins.local.ts:48]: no real need to do async - await here but no big deal

File: packages/common/src/entities/plugins/plugins.local.ts [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+            .unwindPromises();
+
+        await sources
+            .mapEach(async (source) => {
+                const sourcePath = path.join(
+                    this.environment.dirPaths.pluginSources,
+                    `${source.name}${JSON_FILE_EXTENSION}`,
+                );
+
+                await fse.remove(sourcePath);
```
Comment (huboneo) [inline packages/common/src/entities/plugins/plugins.local.ts:67]: no need

File: packages/common/src/models/plugin.model.ts [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+    @IsUrl()
+    public versionsUrl!: string;
+}
+
+export class PluginVersionModel extends ModelAbstract<IPluginVersion> implements IPluginVersion {
+    constructor(props: any) {
+        super(props);
+
+        // reassigning default values doesn't work when it's done from the parent class
```
Comment (huboneo) [inline packages/common/src/models/plugin.model.ts:52]: :)

**Review by huboneo — DISMISSED** [review]
LGTM

**Comment by huboneo** [comment]
Think about the name here, maybe call it `DbmsPlugins` instead?

**Comment by jk05** [comment]
> Think about the name here, maybe call it `DbmsPlugins` instead?

+1

**Review by jk05 — DISMISSED** [review]
Looks good so far

**Review by jk05 — APPROVED** [review]

**Comment by nglgzz** [comment]
[[Kick-off] relate handling DBMS plugins](https://trello.com/c/LDjRztj5/1569-kick-off-relate-handling-dbms-plugins)

**Pull request merged**